Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -170,15 +170,17 @@ public bool Commit(string message, string authorName, string emailAddress)
                     return false;
                 }
 
-                repo.Stage(changes);
+                LibGit2Sharp.Commands.Stage(repo, changes);
                 if (string.IsNullOrWhiteSpace(authorName) ||
                     string.IsNullOrWhiteSpace(emailAddress))
                 {
-                    repo.Commit(message);
+                    var emptyAuthor = new Signature(string.Empty, string.Empty, DateTimeOffset.UtcNow);
```

Comment:
Might be slightly cleaner to just set those vars to empty if they're null, and use single code path.